cpp/src/search/search_ordered.cu

```diff
@@ -68,28 +68,50 @@ std::unique_ptr<column> search_ordered(table_view const& haystack,
 
   auto const comparator = cudf::experimental::row::lexicographic::two_table_comparator(
     matched_haystack, matched_needles, column_order, null_precedence, stream);
-  auto const has_nulls    = has_nested_nulls(matched_haystack) or has_nested_nulls(matched_needles);
-  auto const d_comparator = comparator.less(nullate::DYNAMIC{has_nulls});
+  auto const has_nulls = has_nested_nulls(matched_haystack) or has_nested_nulls(matched_needles);
 
   auto const haystack_it = cudf::experimental::row::lhs_iterator(0);
   auto const needles_it  = cudf::experimental::row::rhs_iterator(0);
 
-  if (find_first) {
-    thrust::lower_bound(rmm::exec_policy(stream),
-                        haystack_it,
-                        haystack_it + haystack.num_rows(),
-                        needles_it,
-                        needles_it + needles.num_rows(),
-                        out_it,
-                        d_comparator);
+  if (cudf::get_nested_columns(haystack).size() > 0 ||
```

Comment:
@jrhemstad suggested a visitor pattern (or similar dispatch) to avoid this repetition of code in the caller. I think this would be a preferable design, because the caller shouldn't have to handle this special case for inputs with nested types. The comparator should do that. Demo: https://godbolt.org/z/ra76jbM78

Reply:
I agree that we need an alternate implementation, but as we discussed offline that is out of scope for this PR, especially because it should be implemented in a way that minimizes divergence between the lexicographic comparator and the equality comparator (and possibly even the hasher, although that's a slightly different can of worms). Let's come back to this after getting this PR merged.

Reply:
Please add this to the follow up task list before resolving.

Reply:
#11667